### PR TITLE
Add eZ Platform Demo v2.5.5

### PR DESCRIPTION
**Pull request by SylvainGuittard**
Template for eZ Platform Demo Open Source Edition (v2.5.5)

### CODE REVIEW

**Comment by SylvainGuittard** [comment]
@Crell could you please have a look? Thanks

**Comment by Crell** [comment]
I'd recommend against including the full 3 part version number here.  That means every time you update it at all this file needs to be updated as well, which is not the workflow we or you want. :smile: Just v2, or v2.5.  For most others we don't even have a version number at all, frankly, which is probably what you want to do here.

File: templates/ezplatform-demo.template.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# project creation time. This is part of the full v2 UI operation mode, which
+# places project schema/options selection early in the creation process, rather
+# than later as it exits now. To allow this schema to be backwards-compatible,
+# this key also gets mapped to the appropriate location in project.settings so
+# that the current UI can have its own workflow overridden as well.
+initialize:
+    repository: git://github.com/ezsystems/ezplatform-demo.git@v2.5.5
+    config: null
+    files: []
+    profile: eZ Platform Demo v2.5.5
```
Comment (Crell) [inline templates/ezplatform-demo.template.yaml]:
```suggestion
    profile: eZ Platform Demo
```

File: templates/ezplatform-demo.template.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# The schema is versioned so that we can establish code paths differently in the future if we need to change this.
+version: 1
+
+# Templates are a small amount of information supporting a template URL. Each template is selectable at the project-creation step.
+info:
+    # Unique machine name, prefaced by a vendor or organization identifier
+    id: ezsystems/ezplatform-demo
+    # The human-readable name of the template.
+    name: eZ Platform Demo 2.5.5
```
Comment (Crell) [inline templates/ezplatform-demo.template.yaml]:
```suggestion
    name: eZ Platform Demo
```

File: templates/ezplatform-demo.template.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# The schema is versioned so that we can establish code paths differently in the future if we need to change this.
+version: 1
+
+# Templates are a small amount of information supporting a template URL. Each template is selectable at the project-creation step.
+info:
+    # Unique machine name, prefaced by a vendor or organization identifier
+    id: ezsystems/ezplatform-demo
```
Comment (Crell) [inline templates/ezplatform-demo.template.yaml]:
```suggestion
    id: ezsystems/ezplatform-demo-v2
```

**Comment by SylvainGuittard** [comment]
@Crell thanks for your review. I have updated the template. I just kept the major version as suggested.
I will update this template when we will release a patch release on version 2.

**Pull request merged**